ext/auto_flush.c

```diff
@@ -11,8 +11,30 @@
 #include "span.h"
 #include "sidecar.h"
 #include "ddshared.h"
+#include "ddappsec.h"
+#include "standalone_limiter.h"
 #include <main/SAPI.h>
 
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
+
+static bool trace_contains_appsec_event(zval *trace) {
+    if (!trace || Z_TYPE_P(trace) != IS_ARRAY) {
+        return false;
+    }
+
+    zval *root_span = zend_hash_index_find(Z_ARR_P(trace), 0);
```

Comment:
I think by the way ddtrace is currently designed, the root span will (almost?) always be the first span.
But that's more by coincidence that the root span just happens to be the last span closed - and spans are serialized in reverse order.
This may or may not change in future via normal refactors. (?)
We should at least note somewhere that we rely on that being the case now.

Reply:
Ok. How would you go approach this then? I'm ok to implement it properly if it does not require any major refactoring

Reply:
I'm not suggesting to actually change it, but primarily to document that you rely on this now.